### PR TITLE
Refactoring and tests

### PR DESCRIPTION
**Pull request by poupotte**
* Separate in several files
   * Add tests
   * Fix icon problem

### CODE REVIEW

File: lib/application.coffee [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+                callback app.icon
+        if not found
+            callback ''
+
+waitInstallComplete = (slug, callback) ->
+    axon   = require 'axon'
+    socket = axon.socket 'sub-emitter'
+    socket.connect 9105
+    noAppListErrMsg = """
+No application listed after installation.
```
Comment (aenario) [inline lib/application.coffee]: Coffeescript """ can be indented (there wont be a \t in the string)
```coffee
     appNotStartedErrMsg = """
           Application is not running after installation.
     """
```

**Comment by aenario** [comment]
Could you replace the Cakefile by a 
    
"scripts":{
   "test":"mocha tests --reporter spec --compilers coffee:coffee-script/register --colors"
}

In the package.json ?
(you will also need to remove the script part from travis.yml)

Otherwise, code wise, everything looks good and it will be way easier to work with the monitor. Thanks.

**Comment by aenario** [comment]
LGTM :+1:

**Comment by frankrousseau** [comment]
Thx to both of you. I can't believe this little too became so big :).

**Pull request merged**